tests/test_models.py

```diff
@@ -83,6 +83,81 @@ def test_model_singlepsr_noise_dip_cusp(nodmx_psrs,caplog):
     x0 = {pname:p.sample() for pname,p in zip(m.param_names, m.params)}
     m.get_lnlikelihood(x0)
 
+def test_model_singlepsr_noise_dm_nondiag(nodmx_psrs,caplog):
```

Comment:
These should be tests of the `'nondiag'` kernels, but the `dmgp_kernel` is set to `'diag'` for all of the tests